### DISCUSSION TITLE
How do you add  Lists to contenttypes from code?

### DISCUSSION THREAD

**Discussion by ghost** [OrchardCMS/OrchardCore#16499] [Q&A] [answered]
I'm current trying to set up a way for users to view contact form submissions in the admin section.  I'm able to create the part and type.  I'm also able to add the part to the type.  However, I can't figure out how to add it as a "list" of "clientmessage" and not just a single "clientmessage"  

this is from my datamigration class:

        public int Create()
        {

            _contentDefinitionManager.AlterPartDefinition(nameof(ClientMessagePart), part => part
                 .Attachable()
            );

            _contentDefinitionManager.AlterTypeDefinition("ClientMessage", type => type
                 .Creatable()
                 .Listable()
                 .WithPart(nameof(ClientMessagePart))
            );

            return 1;
        }

**Comment by Piedone** [comment, marked as answer]
You can create the content types from the admin, then [generate the corresponding migration code](https://github.com/Lombiq/Helpful-Extensions?tab=readme-ov-file#content-definition-code-generation). You can take the Blog and Blog Post content types from the Blog recipe/theme as an inspiration.

**Reply by ghost** [reply]
When I create types through the admin,  is there a way to still have intellisense in visual studio?

**Reply by Piedone** [reply]
If you generate the code as I mentioned then yes, you'll get IntelliSense as usual. However, migrations, as kind of configuration files, will just have code completion for the method calls but not the values you need to pass.
